### DISCUSSION TITLE
post/mock-test-code/

### DISCUSSION THREAD

**Discussion by giscus** [kakaopay/techlog-comments#28] [Announcements]
# post/mock-test-code/

Mock 테스트 코드 작성 중에 마주한 문제들과 그 문제를 해결하는 방법과 노하우를 소개 드립니다.

https://tech.kakaopay.com/post/mock-test-code/

**Comment by Vorschlag-bit** [comment]
msa의 테스트 코드 작성에서 많이 고민이 되었는데, 좋은 방향성을 제시해준 글인 거 같아서 매우 감사합니다.
다만, 위 테스트 코드의 예시는 통신을 feign client가 아닌 건가요? feign client는 자동적으로 primary bean으로 등록되어서 bean 충돌(2개의 primary bean 존재)이 나더라고요

**Reply by cheese10yun** [reply]
> msa의 테스트 코드 작성에서 많이 고민이 되었는데, 좋은 방향성을 제시해준 글인 거 같아서 매우 감사합니다.
다만, 위 테스트 코드의 예시는 통신을 feign client가 아닌 건가요? feign client는 자동적으로 primary bean으로 등록되어서 bean 충돌(2개의 primary bean 존재)이 나더라고요

질문 감사합니다.  
해당 예제에서는 Feign Client 대신 RestTemplate을 사용합니다. 예제에서는 PartnerClient라는 클라이언트 객체를 Bean으로 등록하여 RestTemplate을 감싸는 구조로 구현되어 있기 때문에, Feign Client가 자동으로 primary bean으로 등록되어 발생할 수 있는 bean 충돌 문제는 고려하지 않았습니다.  
즉, 이 예제는 Feign Client를 직접 사용하는 구조가 아니라 RestTemplate 기반의 클라이언트 구현 예시입니다.

**Comment by pak0426** [comment]
점진적으로 테스트 코드에 대한 문제점을 해결하는 과정이 이해가 너무 잘되네요! 감사합니다 잘 읽었습니다
